File: 2016/gara4/regali/att/regali.c
#include <stdio.h>
#include <assert.h>

#define MAXN 10000
#define MAXQ 1000

int compra(int N, int Q, int* G[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int* G[MAXN];

int main() {
    FILE *fr, *fw;
    int N, Q, i, j;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d %d", &N, &Q));
    for(i=0; i<N; i++) {
        G[i] = (int*)malloc(Q*sizeof(int));
        for (j=0; j<Q; j++) {
            assert(1 == fscanf(fr, "%d", &G[i][j]));
        }
    }

    fprintf(fw, "%d\n", compra(N, Q, G));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: 2016/gara4/regali/att/regali.pas
const
    MAXN = 10000;
    MAXQ = 1000;
type
    matrix = array[0..MAXN-1] of array[0..MAXQ-1] of longint;


function compra(N, Q: longint; var V: matrix): longint;
begin
    (* Mettete qui il codice della soluzione *)
    compra := 42;
end;


var
    N, Q, i, j  : longint;
    G           : matrix;
    fr, fw      : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, Q);
    for i:=0 to N-1 do begin
        for j:=0 to Q-1 do
            read(fr, G[i][j]);
        readln(fr);
    end;

    writeln(fw, compra(N, Q, G));
    close(fr);
    close(fw);
end.
